extensions/classify/config.proto

```diff
@@ -0,0 +1,198 @@
+/* Copyright 2020 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = "proto3";
+
+package classify;
+
+import "google/protobuf/duration.proto";
+
+// Configuration to classify http traffic.
+// Traffic classification can be done on mutiple dimensions using request and
+// response headers and other measurements. 
+// The plugin must be configured as request, response or log based on
+// which other plugins need to consume the classfication results.
+// For example: If Stats plugin uses `operation` dimension produced by the classification
+// plugin, then it must be executed in LOG phase *before* the Stats plugin.
+message PluginConfig {
+  // next id: 3
+  // Plugin phase when classification is done.
+  Phase phase = 1;
+
+  // Request classification along multiple dimensions.
+  repeated Dimension dimensions = 2;
+}
+
+// Dimension configuration produces the named dimension with a value based
+// on the match clause in the value. 
+// The value matches are processed sequentially and the first value to match is assigned.
+message Dimension {
+  // For example `operation` is a dimension.
+  string name = 1;
+
+  // At most one of the following values is assigned to the dimension. 
+  repeated Value values = 2;
+}
+
+// Value assigned to the dimension if the request matches.
+message Value {
+  // Assign `value` to the enclosing dimension if match is true.
+  string value = 1;
+
+  // Match specification.
+  // Empty match always matches, which can be used to specify a default value when placed
+  // at the end of the list.
+  Match match = 2;
+}
+
+// Match condition based on request and response information.
+// Every match condition must match for the result to be true.
+message Match {
+  // invert the match condition.
+  bool invert = 1;
+
+  // Match path.
+  repeated StringMatch path = 2;
+
+  // Match scheme: http, https or unset.
+  Scheme scheme = 3;
+
+  // Match method: GET, POST, HEAD, PUT, DELETE, ...
+  repeated string method = 4;
+
+  // Match request headers.
+  // Not implemented.
+  // $hide_from_docs
+  map<string, StringMatch> request_headers = 5;
```

Comment:
Since you have to copy these fields from VirtualService API instead of importing, wdyt about using CEL and listing available attributes?

Have a copy of the API field is kind of a maintenance nightmare with accumulating tech debt of keeping parity.

Reply:
I will try out a CEL only api and see how it works. The proto defn will be more. Concise and kuat and I have been discussing that.
The user experience and validation is the issue.

I am not worried about maintaining parity with virtual service.
Neither APIs matching properties are expected to change often.

Reply:
@kyessenov specifying cel in json with escaping “ will be a pain so we need envoy filter api to do the conversations.